src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

```diff
@@ -0,0 +1,255 @@
+      {
+        switchScores(i, counter);
+      }
+    }
+
+    /// Looks at the first Hit of the given @p id and according to the given @p type ,
+    /// deduces a fitting score and score direction to be switched to.
+    /// Then tries to switch all hits.
+    void switchToGeneralScoreType(ConsensusMap& cmap, ScoreType type, Size& counter, bool unassigned_peptides_too = true)
+    {
+      String new_type = "";
+      for (const auto& f : cmap)
+      {
+        const auto& ids = f.getPeptideIdentifications();
+        if (!ids.empty())
+        {
+          new_type = findScoreType(ids[0], type);
+          if (new_type == ids[0].getScoreType())
+          {
+            return;
+          }
+          else
+          {
+            break;
+          }
+        }
+      }
+
+      if (new_type.empty())
+      {
+        String msg = "First encountered ID does not have the requested score type.";
+        throw Exception::MissingInformation(__FILE__, __LINE__,
+                                            OPENMS_PRETTY_FUNCTION, msg);
+      }
+
+      new_score_ = new_type;
+      new_score_type_ = new_type;
+
+      if (type != ScoreType::RAW && higher_better_ != type_to_better_[type])
+      {
+        OPENMS_LOG_WARN << "Requested non-raw score type does not match the expected score direction. Correcting!\n";
+        higher_better_ = type_to_better_[type];
+      }
+
+      const auto switchScoresSingle = [&counter,this](PeptideIdentification& id){switchScores(id,counter);};
+      cmap.applyFunctionOnPeptideIDs(switchScoresSingle, unassigned_peptides_too);
+    }
+
+
+    /// finds a certain score type in an ID and its metavalues if present, otherwise returns empty string
+    template <typename IDType>
+    String findScoreType(IDType& id, IDScoreSwitcherAlgorithm::ScoreType type)
+    {
+      const String& curr_score_type = id.getScoreType();
+      const set<String>& possible_types = type_to_str_[type];
+      if (possible_types.find(curr_score_type) != possible_types.end())
+      {
+        OPENMS_LOG_INFO << "Requested score type already set as main score: " + curr_score_type + "\n";
+        return curr_score_type;
+      }
+      else
+      {
+        if (id.getHits().empty())
+        {
+          OPENMS_LOG_WARN << "Identification entry used to check for alternative score was empty.\n";
+          return "";
+        }
+        const auto& hit = id.getHits()[0];
+        for (const auto& poss_str : possible_types)
+        {
+          if (hit.metaValueExists(poss_str) || hit.metaValueExists(poss_str + "_score")) return poss_str;
+        }
+        OPENMS_LOG_WARN << "Score of requested type not found in the UserParams of the checked ID object.\n";
+        return "";
+      }
+    }
+
+  private:
+    /// \brief describes a PeptideHit for debug/exception messages
+    /// \param hit the hit
+    /// \return a String description to include in a message
+    String describeHit_(const PeptideHit& hit);
```

Comment:
does it make sense to move this to PeptideHit as operator<<? (same with ProteinHit)